src/scheduler/resv_info.c

```diff
@@ -601,24 +601,24 @@ query_resv(struct batch_status *resv, server_info *sinfo, int *err)
 				}
 			}
 			if (resv_owner == FALSE) {
-				*err = DONT_OWN_RESV;
+				set_schd_error_codes(err, NEVER_RUN, DONT_OWN_RESV);
 				return NULL;
 			}
 			break;
 		}
 	}
 	if (resv_owner == FALSE) {
-		*err = DONT_OWN_RESV;
+		set_schd_error_codes(err, NEVER_RUN, DONT_OWN_RESV);
 		return NULL;
 	}
 
 	if ((advresv = new_resource_resv()) == NULL) {
-		*err = ENOMEM;
+		set_schd_error_codes(err, NEVER_RUN, ERR_SPECIAL);
```

Comment:
If you set this to ERR_SPECIAL, you need to set the argument.  ERR_SPECIAL is saying to not convert the err structure to a static string, but use the generic string inside the err structure itself.

Reply:
As discussed, I have now moved this check to the caller itself